plugins/esi_node_provisioning.py

```diff
@@ -0,0 +1,25 @@
+from rally_openstack.task.scenario import scenario
+from rally import consts
+from rally.task import atomic
+from rally.task import validation
+import subprocess
+import time
+import os
+
+@validation.add("required_platform", platform="openstack", users=True)
+@scenario.configure(name="ScenarioPlugin.esi_node_provisioning")
+
+class EsiNodeProvisioning(scenario.Scenario):
+    def esi_node_provisioning(self, image, network, candidate, resource_class='baremetal', ssh_public_key=None):
+        """Scenario to test 'openstack esi node provisioning' command."""
+        command = f"metalsmith deploy --resource-class {resource_class} --image {image} --network {network} --candidate {candidate} --ssh-public-key {ssh_public_key}"
```

Comment:
I think the `--ssh-public-key` flag should be removed here, since it's added below.

Reply:
Done.